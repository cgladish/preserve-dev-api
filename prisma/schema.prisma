// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id        Int       @id @default(autoincrement())
  email     String    @unique
  username  String    @db.VarChar(255)
  snippets  Snippet[]
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt

  @@index([email, username])
}

model App {
  id        Int       @id @default(autoincrement())
  name      String    @db.VarChar(255)
  snippets  Snippet[]
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
}

model Snippet {
  id                  Int       @id @default(autoincrement())
  public              Boolean   @default(false)
  title               String?   @db.VarChar(255)
  appSpecificDataJson String?   @db.VarChar(255)
  views               Int
  creator             User      @relation(fields: [creatorId], references: [id])
  creatorId           Int
  App                 App       @relation(fields: [appId], references: [id])
  appId               Int
  messages            Message[]
  createdAt           DateTime  @default(now())
  updatedAt           DateTime  @updatedAt
}

model Message {
  id                  Int      @id @default(autoincrement())
  content             String   @db.VarChar(4096)
  sentAt              DateTime
  appSpecificDataJson String?  @db.VarChar(255)
  authorUsername      String   @db.VarChar(255)
  authorIdentifier    String?  @db.VarChar(255)
  authorAvatarUrl     String?  @db.VarChar(255)
  Snippet             Snippet  @relation(fields: [snippetId], references: [id])
  snippetId           Int
}
