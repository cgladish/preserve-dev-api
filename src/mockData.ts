import { User } from "@prisma/client";
import jwt from "jsonwebtoken";
import prisma from "./prisma";

export const testJwtSecret =
  "c9d2ccab48b599c6728b2d1b65bf81d8604be5c05caed49f753f4704e156aa5e";
export const testJwt = jwt.sign(
  {
    sub: "sub1",
    iat: 1665870819,
    exp: 4102444800,
    permissions: ["read:unreviewed", "edit:unreviewed"],
  },
  testJwtSecret,
  { algorithm: "HS512" }
);
export const testOtherJwt = jwt.sign(
  {
    sub: "sub2",
    iat: 1665870819,
    exp: 4102444800,
    permissions: [],
  },
  testJwtSecret,
  { algorithm: "HS512" }
);

export const makeUser = (overwrites: Partial<User> = {}) =>
  prisma.user.create({
    data: {
      username: "crasken",
      displayName: "Crasken",
      sub: "sub1",
      createdAt: new Date(10),
      updatedAt: new Date(10),
      ...overwrites,
    },
  });
