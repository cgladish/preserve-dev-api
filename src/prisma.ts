import { PrismaClient } from "@prisma/client";
import Hashids from "hashids";

const PADDED_ID_LENGTH = 7;

type IdUtils = {
  idToExternalId: (id: number) => string;
  externalIdToId: (externalId: string) => number;
};
export const makeIdUtils = (modelName: string): IdUtils => ({
  idToExternalId: (id: number): string =>
    new Hashids(modelName, PADDED_ID_LENGTH).encode(id),
  externalIdToId: (externalId: string): number => {
    const id = Number(
      new Hashids(modelName, PADDED_ID_LENGTH).decode(externalId)[0]
    );
    if (Number.isNaN(id)) {
      throw new Error("Invalid external ID");
    }
    return id;
  },
});

const prisma = new PrismaClient();
const modelNames: (keyof typeof prisma)[] = [
  "app",
  "comment",
  "message",
  "snippet",
  "user",
];
modelNames.forEach((modelName) =>
  Object.assign(prisma[modelName], makeIdUtils(modelName))
);
export type ExtendedPrismaClient = PrismaClient & {
  app: PrismaClient["app"] & IdUtils;
  comment: PrismaClient["comment"] & IdUtils;
  message: PrismaClient["message"] & IdUtils;
  snippet: PrismaClient["snippet"] & IdUtils;
  user: PrismaClient["user"] & IdUtils;
};
export default prisma as ExtendedPrismaClient;
