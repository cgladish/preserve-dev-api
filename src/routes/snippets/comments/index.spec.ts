import app from "../../../index";
import prisma from "../../../prisma";
import request from "supertest";
import { App, Snippet, User, Comment } from "@prisma/client";
import { makeUser } from "../../../mockData";

describe("comments routes", () => {
  let appEntity: App;
  let creatorEntity: User;
  let snippetEntity: Snippet;

  beforeEach(async () => {
    appEntity = await prisma.app.create({
      data: {
        name: "Discord",
      },
    });
    creatorEntity = await makeUser();
    snippetEntity = await prisma.snippet.create({
      data: {
        appId: appEntity.id,
        public: true,
        title: "Test snippet title",
        appSpecificDataJson: '{"key":"value"}',
        creatorId: creatorEntity.id,
        messages: {
          create: [
            {
              content: "Content",
              sentAt: new Date(10).toISOString(),
              appSpecificDataJson: '{"key2":"value2"}',
              authorUsername: "Icyspawn",
              authorIdentifier: "1234",
              authorAvatarUrl: "http://example.com/123.png",
            },
          ],
        },
      },
    });
  });

  describe("GET /", () => {
    let commentEntities: Comment[];

    beforeEach(async () => {
      await prisma.comment.createMany({
        data: Array(50)
          .fill(null)
          .map((_, index) => ({
            content: `Content ${index}`,
            creatorId: creatorEntity.id,
            snippetId: snippetEntity.id,
            createdAt: new Date(index * 5),
            updatedAt: new Date(index * 10),
          })),
      });
      commentEntities = await prisma.comment.findMany({
        orderBy: { id: "asc" },
      });
    });

    it("can get paginated comments for a snippet", async () => {
      const response = await request(app)
        .get(
          `/snippets/${prisma.snippet.idToExternalId(
            snippetEntity.id
          )}/comments`
        )
        .expect(200);
      expect(response.body).toMatchSnapshot();
    });

    it("can get paginated comments after a cursor for a snippet", async () => {
      const response = await request(app)
        .get(
          `/snippets/${prisma.snippet.idToExternalId(
            snippetEntity.id
          )}/comments`
        )
        .query({
          cursor: prisma.comment.idToExternalId(commentEntities[9].id),
        })
        .expect(200);
      expect(response.body).toMatchSnapshot();
    });
  });
});
