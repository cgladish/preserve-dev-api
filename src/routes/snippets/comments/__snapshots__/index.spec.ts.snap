// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments routes GET / can get paginated comments after a cursor for a snippet 1`] = `
{
  "data": [
    {
      "content": "Content 20",
      "createdAt": "1970-01-01T00:00:00.100Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "PGr7pk2",
      "updatedAt": "1970-01-01T00:00:00.200Z",
    },
    {
      "content": "Content 21",
      "createdAt": "1970-01-01T00:00:00.105Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "0KvAQvG",
      "updatedAt": "1970-01-01T00:00:00.210Z",
    },
    {
      "content": "Content 22",
      "createdAt": "1970-01-01T00:00:00.110Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "O9vp3wW",
      "updatedAt": "1970-01-01T00:00:00.220Z",
    },
    {
      "content": "Content 23",
      "createdAt": "1970-01-01T00:00:00.115Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "6Jw2PrE",
      "updatedAt": "1970-01-01T00:00:00.230Z",
    },
    {
      "content": "Content 24",
      "createdAt": "1970-01-01T00:00:00.120Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "ODwMQwz",
      "updatedAt": "1970-01-01T00:00:00.240Z",
    },
    {
      "content": "Content 25",
      "createdAt": "1970-01-01T00:00:00.125Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "Lmw01v0",
      "updatedAt": "1970-01-01T00:00:00.250Z",
    },
    {
      "content": "Content 26",
      "createdAt": "1970-01-01T00:00:00.130Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "mEkzyrW",
      "updatedAt": "1970-01-01T00:00:00.260Z",
    },
    {
      "content": "Content 27",
      "createdAt": "1970-01-01T00:00:00.135Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "8zkm4rL",
      "updatedAt": "1970-01-01T00:00:00.270Z",
    },
    {
      "content": "Content 28",
      "createdAt": "1970-01-01T00:00:00.140Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "Jgk8awZ",
      "updatedAt": "1970-01-01T00:00:00.280Z",
    },
    {
      "content": "Content 29",
      "createdAt": "1970-01-01T00:00:00.145Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "RorZXwD",
      "updatedAt": "1970-01-01T00:00:00.290Z",
    },
    {
      "content": "Content 30",
      "createdAt": "1970-01-01T00:00:00.150Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "OZwg8vq",
      "updatedAt": "1970-01-01T00:00:00.300Z",
    },
    {
      "content": "Content 31",
      "createdAt": "1970-01-01T00:00:00.155Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "yarDJwR",
      "updatedAt": "1970-01-01T00:00:00.310Z",
    },
    {
      "content": "Content 32",
      "createdAt": "1970-01-01T00:00:00.160Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "40w9Pkx",
      "updatedAt": "1970-01-01T00:00:00.320Z",
    },
    {
      "content": "Content 33",
      "createdAt": "1970-01-01T00:00:00.165Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "jBwlmvL",
      "updatedAt": "1970-01-01T00:00:00.330Z",
    },
    {
      "content": "Content 34",
      "createdAt": "1970-01-01T00:00:00.170Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "6nkJPvY",
      "updatedAt": "1970-01-01T00:00:00.340Z",
    },
    {
      "content": "Content 35",
      "createdAt": "1970-01-01T00:00:00.175Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "XAwVmkW",
      "updatedAt": "1970-01-01T00:00:00.350Z",
    },
    {
      "content": "Content 36",
      "createdAt": "1970-01-01T00:00:00.180Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "68kx2v3",
      "updatedAt": "1970-01-01T00:00:00.360Z",
    },
    {
      "content": "Content 37",
      "createdAt": "1970-01-01T00:00:00.185Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "O2k3avq",
      "updatedAt": "1970-01-01T00:00:00.370Z",
    },
    {
      "content": "Content 38",
      "createdAt": "1970-01-01T00:00:00.190Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "1BkR3wj",
      "updatedAt": "1970-01-01T00:00:00.380Z",
    },
    {
      "content": "Content 39",
      "createdAt": "1970-01-01T00:00:00.195Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "VNkq2w1",
      "updatedAt": "1970-01-01T00:00:00.390Z",
    },
  ],
  "isLastPage": false,
  "totalCount": 50,
}
`;

exports[`comments routes GET / can get paginated comments for a snippet 1`] = `
{
  "data": [
    {
      "content": "Content 0",
      "createdAt": "1970-01-01T00:00:00.000Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "z9weAwy",
      "updatedAt": "1970-01-01T00:00:00.000Z",
    },
    {
      "content": "Content 1",
      "createdAt": "1970-01-01T00:00:00.005Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "2NrBMvP",
      "updatedAt": "1970-01-01T00:00:00.010Z",
    },
    {
      "content": "Content 2",
      "createdAt": "1970-01-01T00:00:00.010Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "x9wWnkO",
      "updatedAt": "1970-01-01T00:00:00.020Z",
    },
    {
      "content": "Content 3",
      "createdAt": "1970-01-01T00:00:00.015Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "lWkQXr2",
      "updatedAt": "1970-01-01T00:00:00.030Z",
    },
    {
      "content": "Content 4",
      "createdAt": "1970-01-01T00:00:00.020Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "PzkX7rb",
      "updatedAt": "1970-01-01T00:00:00.040Z",
    },
    {
      "content": "Content 5",
      "createdAt": "1970-01-01T00:00:00.025Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "KekOlwN",
      "updatedAt": "1970-01-01T00:00:00.050Z",
    },
    {
      "content": "Content 6",
      "createdAt": "1970-01-01T00:00:00.030Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "dXkj8vl",
      "updatedAt": "1970-01-01T00:00:00.060Z",
    },
    {
      "content": "Content 7",
      "createdAt": "1970-01-01T00:00:00.035Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "9dwbAk3",
      "updatedAt": "1970-01-01T00:00:00.070Z",
    },
    {
      "content": "Content 8",
      "createdAt": "1970-01-01T00:00:00.040Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "ebkPmwz",
      "updatedAt": "1970-01-01T00:00:00.080Z",
    },
    {
      "content": "Content 9",
      "createdAt": "1970-01-01T00:00:00.045Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "NOkGWkW",
      "updatedAt": "1970-01-01T00:00:00.090Z",
    },
    {
      "content": "Content 10",
      "createdAt": "1970-01-01T00:00:00.050Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "xBknLvK",
      "updatedAt": "1970-01-01T00:00:00.100Z",
    },
    {
      "content": "Content 11",
      "createdAt": "1970-01-01T00:00:00.055Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "1zw60rq",
      "updatedAt": "1970-01-01T00:00:00.110Z",
    },
    {
      "content": "Content 12",
      "createdAt": "1970-01-01T00:00:00.060Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "GEwYnvo",
      "updatedAt": "1970-01-01T00:00:00.120Z",
    },
    {
      "content": "Content 13",
      "createdAt": "1970-01-01T00:00:00.065Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "1zv5Rr4",
      "updatedAt": "1970-01-01T00:00:00.130Z",
    },
    {
      "content": "Content 14",
      "createdAt": "1970-01-01T00:00:00.070Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "l8kNbkK",
      "updatedAt": "1970-01-01T00:00:00.140Z",
    },
    {
      "content": "Content 15",
      "createdAt": "1970-01-01T00:00:00.075Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "J5rd9ke",
      "updatedAt": "1970-01-01T00:00:00.150Z",
    },
    {
      "content": "Content 16",
      "createdAt": "1970-01-01T00:00:00.080Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "WVk4Dke",
      "updatedAt": "1970-01-01T00:00:00.160Z",
    },
    {
      "content": "Content 17",
      "createdAt": "1970-01-01T00:00:00.085Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "Gdvy8wq",
      "updatedAt": "1970-01-01T00:00:00.170Z",
    },
    {
      "content": "Content 18",
      "createdAt": "1970-01-01T00:00:00.090Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "DKrEowP",
      "updatedAt": "1970-01-01T00:00:00.180Z",
    },
    {
      "content": "Content 19",
      "createdAt": "1970-01-01T00:00:00.095Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "JAk1Bw2",
      "updatedAt": "1970-01-01T00:00:00.190Z",
    },
  ],
  "isLastPage": false,
  "totalCount": 50,
}
`;

exports[`comments routes GET / finds last page correctly 1`] = `
{
  "data": [
    {
      "content": "Content 40",
      "createdAt": "1970-01-01T00:00:00.200Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "PzrK8kN",
      "updatedAt": "1970-01-01T00:00:00.400Z",
    },
    {
      "content": "Content 41",
      "createdAt": "1970-01-01T00:00:00.205Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "oNvaBrA",
      "updatedAt": "1970-01-01T00:00:00.410Z",
    },
    {
      "content": "Content 42",
      "createdAt": "1970-01-01T00:00:00.210Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "AlwojkK",
      "updatedAt": "1970-01-01T00:00:00.420Z",
    },
    {
      "content": "Content 43",
      "createdAt": "1970-01-01T00:00:00.215Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "jrLjlwy",
      "updatedAt": "1970-01-01T00:00:00.430Z",
    },
    {
      "content": "Content 44",
      "createdAt": "1970-01-01T00:00:00.220Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "9weAAwy",
      "updatedAt": "1970-01-01T00:00:00.440Z",
    },
    {
      "content": "Content 45",
      "createdAt": "1970-01-01T00:00:00.225Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "NrBNMrP",
      "updatedAt": "1970-01-01T00:00:00.450Z",
    },
    {
      "content": "Content 46",
      "createdAt": "1970-01-01T00:00:00.230Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "9wWdnvO",
      "updatedAt": "1970-01-01T00:00:00.460Z",
    },
    {
      "content": "Content 47",
      "createdAt": "1970-01-01T00:00:00.235Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "WkQgXv2",
      "updatedAt": "1970-01-01T00:00:00.470Z",
    },
    {
      "content": "Content 48",
      "createdAt": "1970-01-01T00:00:00.240Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "zkXo7rb",
      "updatedAt": "1970-01-01T00:00:00.480Z",
    },
    {
      "content": "Content 49",
      "createdAt": "1970-01-01T00:00:00.245Z",
      "creator": {
        "createdAt": "1970-01-01T00:00:00.010Z",
        "displayName": "Crasken",
        "id": "E21K9jm",
        "username": "crasken",
      },
      "id": "ekOVlrN",
      "updatedAt": "1970-01-01T00:00:00.490Z",
    },
  ],
  "isLastPage": true,
  "totalCount": 50,
}
`;

exports[`comments routes POST / can create a comment 1`] = `
{
  "content": "contented",
  "creator": {
    "createdAt": "1970-01-01T00:00:00.010Z",
    "displayName": "Crasken",
    "id": "E21K9jm",
    "username": "crasken",
  },
  "id": "z9weAwy",
}
`;

exports[`comments routes POST / can create a comment 2`] = `
{
  "content": "contented",
  "creatorId": 1,
  "id": 1,
  "snippetId": 1,
}
`;
